docs/reference.md

```diff
@@ -303,6 +311,71 @@ Specify the config file to write to. Without this option, your configuration
 will be written to `~/.gemstash/config.yml`. If you write to a custom location,
 you will need to pass the `--config-file` option to all Gemstash commands.
 
+## Preload
+
+Preload all the gems in your gemstash server from the default upstream.
+This can be useful if you plan to go somewhere where the internet connection
+will not be reliable or available.
+It is worth noting that this will require a large amount of disk space and
+computing time.
+
+### Usage
+
+```
+gemstash preload
+gemstash preload --latest
+gemstash preload --limit=1000
+gemstash preload --limit=1000 --skip=1000
+gemstash preload --threads=10 --server_url=http://my-server-url:9292
+```
+
+### Options
+
+#### --latest
+
+**Usage:** `--latest`
+
+**Description**<br />
+Only fetch the latest specs. This will limit the amount of disk space required,
+but there is no guarantee that the gems you are using will have all their
+dependencies met.
+
+#### --skip
+
+**Usage:** `--skip=<number>`
+
+**Description**<br />
+Number of gems to skip when preloading. This can be useful to load in chunks or
+to resume a gem preloading.
+
+#### --limit
+
+**Usage:** `--limit=<number>`
+
+**Description**<br />
+Limit for the number of gems to preload. In conjunction with [--skip](#--skip)
+this can be used for loading gems in chunks or resuming the process.
```

Comment:
I think I wouldn't worry about this. If you ever get into the case of "I want to fetch all the gems" you would probably just run agains all the gems and let it be.

If you need to make a second run, you have to consider that gem stash will not make a new http request to ruby gems as the gems are already stored locally in gem stash, so those calls will only go as far as loading the properties file, which is not a huge impact.

So I don't think we need to worry too much about this, at least not for now.

Still I will run some tests to see how much of a difference it is to load, say, 50,000 gems and then retry again. Just for curiosity.